cmd/crc/cmd/delete.go

```diff
@@ -87,3 +93,49 @@ func (s *deleteResult) prettyPrintTo(writer io.Writer) error {
 	}
 	return nil
 }
+
+// TODO: remove it after we do one more upgrade of the podman bundle from 4.4.1
+func removeCRCHostEntriesFromKnownHosts() error {
+	knownHostsPath := filepath.Join(constants.GetHomeDir(), ".ssh", "known_hosts")
+	f, err := os.Open(knownHostsPath)
+	if err != nil {
+		return fmt.Errorf("Unable to open user's 'known_hosts' file: %w", err)
+	}
+	defer f.Close()
+
+	tempHostsFile, err := os.CreateTemp("", "crc")
+	if err != nil {
+		return fmt.Errorf("Unable to create temp file: %w", err)
+	}
+	defer func() {
+		os.Remove(tempHostsFile.Name())
+		tempHostsFile.Close()
```

Comment:
shouldn't we first close the file and the remove?

Reply:
right, the order is wrong, thanks

Reply:
For what it's worth, I don't think the order matters. 
After `os.Remove(tempHostsFile.Name())` I'd expect that it's still possible to even read/write to the file even if it's no longer visible in the file system. Read/Write/Close act on an open file descriptor, which is valid until it's closed. Remove acts on a filename, which is not directly related to the file descriptor for this file.

Reply:
hmm, i was also thinking if the last `os.Rename()` would be problematic as that also happens before the close, good to know!

Reply:
On windows this order matter, In the past we had issue when we tried to delete a file without closing it file descriptor and getting error. On unix/linux side order doesn't matter.

Reply:
yep, on windows got the error, now closing the files before the rename, thx